src/app/shared/search/search.component.ts

```diff
@@ -332,6 +349,15 @@ export class SearchComponent implements OnInit {
     this.sidebarService.expand();
   }
 
+  /**
+   * Emit event to refresh filter content
+   * @param $event
+   */
+  public onContentChange($event: any) {
+    this.retrieveFilters(this.lastSearchOptions);
+    this.refreshFilters.next(true);
```

Comment:
IMO it's most likely that the issue with filter facets not refreshing starts somewhere in this component, and isn't directly related to dynamic loading.

I thought we could address it by pushing to `refreshFilters` within the subscription in `retrieveFilters` but this isn't sufficient.